javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/EndpointCharacteristics.qll

```diff
@@ -108,3 +117,317 @@ private class NosqlInjectionSinkCharacteristic extends EndpointCharacteristic {
     confidence = 1.0
   }
 }
+
+/*
+ * Characteristics that are indicative of not being a sink of any type.
+ */
+
+/**
+ * A characteristic that is an indicator of not being a sink of any type, because it's an argument that has a manual
+ * model.
+ */
+abstract private class OtherModeledArgumentCharacteristic extends EndpointCharacteristic {
```

Comment:
For now, I've created a hierarchy of characteristics that closely reflects the hierarchy of classes and predicates in the old implementation. I'm happy to change this hierarchy if there's a different one that makes more sense. This would require JS expertise I don't have, so for now I followed the old implementation.